### PR TITLE
$set and $unset operation support for vector data field

### LINKED ISSUE #457: Update command changes for vector search
Update operations will be changed to add the vector values into query_vector_value if present.
Update clause to support $set and $unset operator for $vector field.

API: /v1/{namespace}/{collection}   
Command: UpdateOne for set
```
{
  "updateOne": {
    "filter" : {"_id" : "update_doc1"},
    "update" : {"$set" : {"$vector": [0.0122,0.2344,....]}}
   }
}
```

API: /v1/{namespace}/{collection}   
Command: UpdateOne for unset
```
{
  "updateOne": {
    "filter" : {"_id" : "update_doc1"},
    "update" : {"$unset" : {"$vector": null}}
   }
}
```

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
$set and $unset operation support for vector data field

**Which issue(s) this PR fixes**:
Fixes #457 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateOperation.java [hunk 1]
```diff
@@ -50,6 +50,13 @@ public boolean shouldApplyIf(boolean isInsert) {
    * specifically Document's primary id, {@code _id}.
    */
   protected static String validateUpdatePath(UpdateOperator oper, String path) {
+    if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(path)
+        && !(oper.operator().equals("$set") || oper.operator().equals("$unset"))) {
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateOperation.java:54]: Can we also support `$setOnInsert` for `$vector`? Not a huge deal, but I don't see any reason why we can't support `$setOnInsert`.

**Review by vkarpov15 — COMMENTED** [review]
One small comment, otherwise LGTM so far

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java [hunk 2]
```diff
@@ -318,6 +319,65 @@ public void insertVectorSearch() {
     }
   }
 
+  public void setUpData() {
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:322]: Maybe use more descriptive name like "insertVectorDocuments()"?

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateOperation.java (hunk 1 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateOperation.java:54]: OK will add $setOnInsert operation for vector search

**Pull request merged**